projects/media-viewer/src/lib/store/reducers/icp.reducer.ts

```diff
@@ -0,0 +1,47 @@
+import * as fromIcpActions from '../actions/icp.action';
+import { IcpSession, IcpState } from '../../icp/icp.interfaces';
+
+export const initialIcpSessionState: IcpState = {
+  session: null,
+  presenter: null,
+  client: null
+};
+
+export function icpReducer (state = initialIcpSessionState,
+                                  action: fromIcpActions.IcpActions): IcpState {
+
+  switch (action.type) {
+
+    case fromIcpActions.ICP_SOCKET_SESSION_JOINED: {
+      const session: IcpSession = {...action.payload.session};
```

Comment:
if you need to do this {...} then is better to decouple it in effect so that mutable objects are not sent in action